File: src/TwoSum.java
import java.util.HashMap;
// Time Complexity : O(N)
// Space Complexity : O(N)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no

public class TwoSum {
        public int[] twoSum(int[] nums, int target) {
            HashMap<Integer, Integer> hash = new HashMap<>();
            for(int i=0; i<nums.length; i++) {
                if(!hash.containsKey(nums[i])) {
                    hash.put(target-nums[i], i);
                }
                else {
                    return new int[]{hash.get(nums[i]), i};
                }
            }
            return nums;
        }

}

File: twoSum.js
// Time Complexity : O(n) n is the length of array nums
// Space Complexity : O(n) n is the length of array nums
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    // If nums is invalid or is Empty
    if(nums===null || nums.length===0)
        return [-1, -1];
    
    // Map to store the number and its index
    let map = new Map();
    for(let i=0; i<nums.length; i++){
        let num = nums[i];
        let diff = target - num;
        if(map.has(diff)){
            return [i, map.get(diff)];
        } else {
            map.set(num, i);
        }
    }
    return [-1, -1];
    
};
